consuldemocracy/consuldemocracy
# Add CSV Download Feature for Proposals, Debates, and Polls

### Summary
This pull request introduces a feature that allows administrators to download CSV files containing all records from three modules: Proposals, Debates, and Polls. The feature is accessible through individual actions for each module and generates a CSV file with relevant data for all records.

> Note: This is work in progress. At the moment only Proposals module has been implemented as a first approach. Please, comment if the way this is done is ok for you. If this receive the approval from the team, it will be implemented for Debates and Polls.

### Key features
1. Download CSV for Proposals
Added a `download_csv` action in Admin::ProposalsController.
Generates a CSV file with headers and data columns relevant to Proposals (id, title, author email, summary, etc.).

2. Exporter model
For Proposal it has been created a new Exporter dedicated only to Proposals so there can be customized the fields and formats requested for the document.

3. Routing
Added download_csv routes under the respective namespace.

4. View Integration
Added "Download Proposals" links to the index pages for Proposals.

5. CSV File Naming
File names are the module name (e.g., proposals.csv).

### How to Test
Navigate to the Admin Panel for Proposals.
Click the "Download CSV" link on the index page.
Verify that:
- A CSV file is downloaded.
- The fields are relevant and sufficient.
- The file contains the correct data, sorted by creation date (oldest to newest).

### Future Enhancements
Add filtering capabilities to allow downloading specific subsets of data (e.g., by date range or status).
Explore generating CSVs asynchronously for large datasets.
